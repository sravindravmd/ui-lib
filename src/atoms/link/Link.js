import React from 'react'
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'

import theme from '../../shared/theme'

const StyledLink = styled.a`
  font-family: ${theme.typography.link.fontFamily};
  font-size: ${theme.typography.link.fontSize};
  line-height: ${theme.typography.link.lineHeight};
  color: ${theme.palette.ui.cta.blue};
  text-align: center;
  outline: none;
  text-decoration: none;
  cursor: pointer;
  border: 1px solid transparent;
  padding: 0px 2px;
  :hover {
    color: ${theme.palette.ui.neutral.grey2};
  }
  :focus {
    padding: 0px 2px;
    border: 1px solid ${theme.palette.ui.cta.yellow};
    border-radius: 6px;
  }
  :active {
    border: 1px solid transparent;
    color: ${theme.palette.ui.neutral.grey1};
  }
  ${({ disabled }) =>
    disabled &&
    css`
      color: ${theme.palette.brand.primary.charcoal};
      opacity: 0.4;
      pointer-events: none;
    `}
`

export const Link = React.forwardRef(({ text, href, ...props }, ref) => {
  return (
    <StyledLink {...props} href={href} ref={ref}>
      {text}
    </StyledLink>
  )
})

Link.propTypes = {
  text: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
}
