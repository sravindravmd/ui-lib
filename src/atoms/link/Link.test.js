import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Link } from './Link'

afterEach(cleanup)

describe('<Link/>', () => {
  it('renders correctly', () => {
    const { container } = render(<Link text='I am a Link' href='/' />)
    expect(container).toMatchSnapshot()
  })

  it('render prop values correctly', () => {
    const { getByText } = render(<Link text='I am a Link' href='/' />)
    const link = getByText('I am a Link')
    expect(link).toHaveTextContent('I am a Link')
    expect(link.getAttribute('href')).toBe('/')
  })
})
