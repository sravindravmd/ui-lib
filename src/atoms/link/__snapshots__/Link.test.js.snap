// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link/> renders correctly 1`] = `
<div>
  <a
    class="sc-bdfBwQ gWHcFY"
    href="/"
  >
    I am a Link
  </a>
</div>
`;
